### PR TITLE
Fix for iOS 11

### PR DESCRIPTION
**Pull request by naveen-c**
For changes to navigationBar and scrollView contentInsets

### CODE REVIEW

**Review by anirudhramanan — COMMENTED** [review]
Changes looks good to me. Did we test this on iOS 9 and 10 devices ?

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift [hunk 1]
```diff
@@ -54,9 +54,8 @@ class HidingNavExtensionViewController: UIViewController, UITableViewDataSource,
 	// MARK: UITableViewDelegate
 	
 	func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
-		hidingNavBarManager?.shouldScrollToTop()
-		
-		return true
```
Comment (anirudhramanan) [inline HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift:59]: We can use nil coalescing operator here.

return hidingNavBarManager?.shouldScrollToTop() ?? true

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavTabViewController.swift [hunk 2]
```diff
@@ -56,9 +56,8 @@ class HidingNavTabViewController: UIViewController, UITableViewDataSource, UITab
 	// MARK: UITableViewDelegate
 	
 	func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
-		hidingNavBarManager?.shouldScrollToTop()
-		
-		return true
+        guard let shouldScroll = hidingNavBarManager?.shouldScrollToTop() else { return true }
```
Comment (anirudhramanan) [inline HidingNavigationBarSample/HidingNavigationBarSample/HidingNavTabViewController.swift]: Same

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavToolbarViewController.swift [hunk 3]
```diff
@@ -50,9 +50,8 @@ class HidingNavToolbarViewController: UIViewController, UITableViewDataSource, U
 	// MARK: UITableViewDelegate
 	
 	func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
-		hidingNavBarManager?.shouldScrollToTop()
-		
-		return true
+        guard let shouldScroll = hidingNavBarManager?.shouldScrollToTop() else { return true }
+        return shouldScroll
```
Comment (anirudhramanan) [inline HidingNavigationBarSample/HidingNavigationBarSample/HidingNavToolbarViewController.swift]: Same

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavViewController.swift [hunk 4]
```diff
@@ -46,9 +46,8 @@ class HidingNavViewController: UIViewController, UITableViewDataSource, UITableV
 	// MARK: UITableViewDelegate
 	
 	func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
-		hidingNavBarManager?.shouldScrollToTop()
-		
-		return true
+        guard let shouldScroll = hidingNavBarManager?.shouldScrollToTop() else { return true }
+        return shouldScroll
```
Comment (anirudhramanan) [inline HidingNavigationBarSample/HidingNavigationBarSample/HidingNavViewController.swift]: Same

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift (hunk 1 above)
Comment (naveen-c) [reply HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift:59]: @anirudhramanan updated

**Comment by naveen-c** [comment]
@anirudhramanan Tested on simulators, not on device. Will do that now

**Comment by anirudhramanan** [comment]
okay. We can bump the pod version once done

**Comment by naveen-c** [comment]
Verified on iOS 10, iOS 11 devices

File: HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift (hunk 1 above)
Comment (naveen-c) [reply HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift:59]: Also need to mention this change in notes for new pod version

Comment (anirudhramanan) [reply HidingNavigationBarSample/HidingNavigationBarSample/HidingNavExtensionViewController.swift:59]: yes

**Comment by anirudhramanan** [comment]
iOS 9 ? These changes should work, merging this, we can test it on iOS9 after bumping the pod

**Review by anirudhramanan — APPROVED** [review]

**Pull request merged**